satellite/metainfo/service.go

```diff
@@ -49,15 +49,22 @@ func (s *Service) Put(ctx context.Context, path string, pointer *pb.Pointer) (er
 	return Error.Wrap(err)
 }
 
-// UpdatePieces atomically adds toAdd pieces and removes toRemove pieces from
+// UpdatePieces calls UpdatePiecesCheckDuplicates with checkDuplicates equal to false.
+func (s *Service) UpdatePieces(ctx context.Context, path string, ref *pb.Pointer, toAdd, toRemove []*pb.RemotePiece) (pointer *pb.Pointer, err error) {
+	return s.UpdatePiecesCheckDuplicates(ctx, path, ref, toAdd, toRemove, false)
+}
+
+// UpdatePiecesCheckDuplicates atomically adds toAdd pieces and removes toRemove pieces from
 // the pointer under path. ref is the pointer that caller received via Get
 // prior to calling this method.
 //
-// It will first check if the pointer has been deleted or replaced. Then it
-// will remove the toRemove pieces and then it will add the toAdd pieces.
+// It will first check if the pointer has been deleted or replaced.
+// Then if checkDuplicats is true it will return an error if the nodes to be
+// added already have a piece for the pointer.
```

Comment:
nit - `...if the nodes to be added are already in the pointer.`

Reply:
Updated